server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -813,6 +834,7 @@ public void executeQueryPhase(
      * when the object is no longer needed.
      */
     public void executeQueryPhase(QuerySearchRequest request, SearchShardTask task, ActionListener<QuerySearchResult> listener) {
+        listener = maybeWrapListenerForStackTrace(listener);
```

Comment:
I can see how we need end-to-end tests for this functionality, yet I wish we had some unit tests of some sort as well. The bits in SearchServiceTests are an example. I know that it takes effort to create a full-blown SearchService instance, but we could probably test the wrapping at least in isolation, and maybe the fact that it is called in SearchServiceTests (which is a single node test class)

Reply:
I've added also a unit test for the listener wrapping